services/passport.js

```diff
@@ -15,11 +15,15 @@ passport.use(
         dbService.getUserByGoogleID(profile.id, function(result) {
             // If we've found the user via googleID, then she's already OAuthed.
             if (result && result.length != 0 && result[0]) {
+                // Trace it when the user logs in
+                dbService.insertActivityLog(result[0].ID, result[0].FirstName, result[0].LastName, "Existing user finished OAuth and logged in");                
+
                 return done(null, result[0].ID);                
             }
             // If the user isn't OAuthed yet, insert the User model to our DB
             else {
                 dbService.insertUser(profile, function(id) {
+                    dbService.insertActivityLog(id, null, null, "New user OAuth'ed and logged in");                                    
                     done(null, id);
```

Comment:
Note here that Firstname and Lastname are null. We do not have access to these. When we insert the user, we only get back the user's ID.

I don't want to make another DB call to get the user via SQL and we could always find this user's name through the inserted ID.

Another alternate would be to use the "profile" variable for the first and last name but I'm not sure if that's 100% reliable or accurate. What if Google doesn't provide us their first or last names? there'd be another exception.

Reply:
Do first and last names need to be associated with the logs? The user id alone is enough IMO if you just want to associate a user with an action

Reply:
Ye, but it makes it more usable for people who look at the logs. Seeing a name would be helpful.

EX: "Oh, looks like Minh DID sign in but then logged out"

It elminates the step of having to look up who logged in with the ID.

Reply:
@trevorhalvorson If this is good enough lemme know

Reply:
@MinhLu12 I still don't see the value in knowing their full name. What's the use case where that is needed?